chapters/sampling-randomization-power.tex

```diff
@@ -1,386 +1,428 @@
-at the level at which the treatment was assigned.\sidenote{\url{https://blogs.worldbank.org/impactevaluations/when-should-you-cluster-standard-errors-new-wisdom-econometrics-oracle}}
-
-Clustered randomization must typically be implemented manually;
-it typically relies on subsetting the data intelligently
-to the desired assignment levels.
-We demonstrate here.
-
-\codeexample{randtreat-clusters.do}{./code/randtreat-clusters.do}
-
-%------------------------------------------------
-\section{Power calculations}
-
-When we have decided on a practical sampling and randomization design,
-we next assess its \textbf{power}.\sidenote{\url{https://dimewiki.worldbank.org/wiki/Power_Calculations_in_Stata}}
-\index{power}
-Statistical power can be described in a few ways,
-each of which has different uses.\sidenote{\url{http://www.stat.columbia.edu/~gelman/stuff_for_blog/chap20.pdf}}
-The purpose of power calculations is not to
-demonstrate that a study is ``strong'',
-but rather to identify where the strengths and weaknesses
-of your design are located, so that readers
-can correctly assess the evidentiary value of
-any results (or null results) in the analysis.
-This should be done before going to the field,
-across the entire range of research questions
-your study might try to answer,
-so you know the relative tradeoffs you will face
-by changing your sampling and randomization schemes
-and can select your final strategy appropriately.
-
-The classic definition of power is
-``the likelihood that your design detects a significant treatment effect,
-given that there is a non-zero true effect in reality''.
-Here we will look at two useful practical applications
-of that definition and show what quantitative results can be obtained.
-We suggest doing all power calculations by simulation;
-you are very unlikely to be able to determine analytically
-the power of your study unless you have a very simple design.
-Stata has some commands that can calculate power for
+but the study is interested in household behavior.
+This type of structure is called \textbf{clustering},\sidenote{
+  \url{https://dimewiki.worldbank.org/wiki/Multi-stage_(Cluster)_Sampling}}
+and the groups in which units are assigned to treatment are called clusters.
+The same principle extends to sampling:
+it may be infeasible to decide whether to test individual children
+within a single classroom, for example.
+
+Clustering is procedurally straightforward in Stata,
+although it typically needs to be performed manually.
+To cluster sampling or randomization,
+\texttt{preserve} the data, keep one observation from each cluster
+using a command like \texttt{bys [cluster] : keep if \_n == 1}.
+Then sort the data and set the seed, and generate the random assignment you need.
+Save the assignment in a separate dataset or a \texttt{tempfile},
+then \texttt{restore} and \texttt{merge} the assignment back on to the original dataset.
```

Comment:
I disagree that this is the best way to do either sampling or randomization using clusters. I think we should have one master data set per unit of observation. So there should be a classroom/village level master data set, and that is the data set that will be used for sampling or randomization. It will then be merged 1:m to the student/household level data set, using the necessary variables.

Reply:
I agree, I think the more language agnostic way to express this is to tell the reader to:

_create a data set where each cluster unit is an observation (unless that already exist), and randomize on that data set, and then merge back the results_.

I know that is what you are saying Ben, but I think introducing a specific code in the middle of the text I think just confuse the reader.

Reply:
7496599be200a565ffae8a3b8b88b36acbd6febd

Reply:
good for me. @luizaandrade , you started this comment so you get to resolve it